doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,679 @@
+
+        { "command": { "action":"suggestions", "source": "tasks" } },
+        { "command": { "action":"suggestions", "source": "local" } },
+
+        { "command": { "action":"suggestions", "source": ["local", "tasks", "commandHistory"] } },
+
+        { "command": { "action":"suggestions", "source": "Microsoft.Terminal.Extensions.BufferComplete" } },
+```
+
+Each of these `suggestions` actions would open the Suggestions UI with a
+different set of actions.
+
+* `commandHistory`: Use commands from this session, as identified via shell
+  integration. This won't be able to return any suggestions if the user has not
+  configured their shell to support shell integration sequences yet.
+* `directoryHistory`: Populate the list with a series of `cd {path}` commands,
+  where the paths are populated via shell integration. Paths are in MRU order.
+* `tasks`: Populate the list with all `sendInput` actions in the user's settings
+  file. The command structure should remain unchanged. For example, if they have
+  `sendInput` actions nested under a "git" command, then the "git" entry will
+  remain in this tasks view with their `sendInput` actions nested inside it. For
+  more details, see the [Tasks] spec.
+* `local`: Populate the list with tasks that are located in the CWD, in a file
+  named `.wt.json`. For more details, see the [Tasks] spec.
+* `Microsoft.Terminal.Extensions.BufferComplete`: As an example, this
+  demonstrates how an action might be authored to reference a suggestion source
+  from an extension[[1](#footnote-1)].
+
+Each of these different sources will build a different set of `Command`s,
+primarily populated with `sendInput` actions. We'll load those `Command`s into
+the Suggestions UI control, and open it at the text cursor.
+
+To drill in on a single example - the `commandHistory` source. In that
+particular case, the TerminalPage will query the active TermControl for a list
+of its recent commands. If it knows these (via shell integration), then the
+TerminalPage will use that list of commands to build a list of `sendInput`
+actions. Those will then get fed to the suggestions UI.
+
+Not listed above is [shell-driven autocompletion]. These aren't something that
+the Terminal can invoke all on its own - these are something the shell would
+need to invoke themselves.
+
+#### Pre-populate the current commandline context
+
+Consider the following scenario. A user has typed `git c` in their shell, and
+has [shell integration] enabled for their shell. They want to open the
+Suggestions UI filtered to their recent history, but starting with what they've
+already typed. To support this scenario, we'll add an additional property:
+
+* `"useCommandline"`: `bool`
+  * `true`: the current commandline the user has typed will pre-populate the
+    filter of the Suggestions UI. This requires that the user has enabled shell
+    integration in their shell's config.
+  * `false`: the filter will start empty.

```

Comment:
I'll try and noodle on a reason you could _possibly_ want `false`. It's so easy to just <kbd>ctrl+bksp</kbd> if you _don't_ want the suggestion too

Reply:
This might be an artifact of organic development here. I think with the way the sxnui landed, `useCommandline: false` almost never makes sense. 

It might have in a more historical context where I was like, pre-filtering the results before they even get to the sxnui, and the filter text was applied after that.

Reply:
ah wait it was for 

```json
{ "command": { "action":"suggestions", "source": ["all"], "useCommandline": false, "nesting": "source" } },
```

because `git c` when filtering a list of [Tasks..., Recent commands..., Recent directories..., Docker..., Git...] was useless.